pkg/cmd/server/start/master_args.go

```diff
@@ -51,6 +52,9 @@ type MasterArgs struct {
 	StartControllers bool
 	PauseControllers bool
 
+	// DNSDomain is the authoritative domain of the DNS server.
```

Comment:
I actually would not expect to expose this as a command line option. As an advanced option when running from config, it seems more reasonable. The command line args cover a fraction of the config possibilities, designed to cover the most common launch cases.